// @aws-sdk/client-elastic-beanstalk/dist-es/models/enums.js
export const ActionHistoryStatus = {
    Completed: "Completed",
    Failed: "Failed",
    Unknown: "Unknown",
};
export const ActionStatus = {
    Pending: "Pending",
    Running: "Running",
    Scheduled: "Scheduled",
    Unknown: "Unknown",
};
export const ActionType = {
    InstanceRefresh: "InstanceRefresh",
    PlatformUpdate: "PlatformUpdate",
    Unknown: "Unknown",
};
export const SourceRepository = {
    CodeCommit: "CodeCommit",
    S3: "S3",
};
export const SourceType = {
    Git: "Git",
    Zip: "Zip",
};
export const ApplicationVersionStatus = {
    Building: "Building",
    Failed: "Failed",
    Processed: "Processed",
    Processing: "Processing",
    Unprocessed: "Unprocessed",
};
export const EnvironmentHealth = {
    Green: "Green",
    Grey: "Grey",
    Red: "Red",
    Yellow: "Yellow",
};
export const EnvironmentHealthStatus = {
    Degraded: "Degraded",
    Info: "Info",
    NoData: "NoData",
    Ok: "Ok",
    Pending: "Pending",
    Severe: "Severe",
    Suspended: "Suspended",
    Unknown: "Unknown",
    Warning: "Warning",
};
export const EnvironmentStatus = {
    Aborting: "Aborting",
    Launching: "Launching",
    LinkingFrom: "LinkingFrom",
    LinkingTo: "LinkingTo",
    Ready: "Ready",
    Terminated: "Terminated",
    Terminating: "Terminating",
    Updating: "Updating",
};
export const ComputeType = {
    BUILD_GENERAL1_LARGE: "BUILD_GENERAL1_LARGE",
    BUILD_GENERAL1_MEDIUM: "BUILD_GENERAL1_MEDIUM",
    BUILD_GENERAL1_SMALL: "BUILD_GENERAL1_SMALL",
};
export const ConfigurationDeploymentStatus = {
    deployed: "deployed",
    failed: "failed",
    pending: "pending",
};
export const PlatformStatus = {
    Creating: "Creating",
    Deleted: "Deleted",
    Deleting: "Deleting",
    Failed: "Failed",
    Ready: "Ready",
};
export const ConfigurationOptionValueType = {
    List: "List",
    Scalar: "Scalar",
};
export const EnvironmentHealthAttribute = {
    All: "All",
    ApplicationMetrics: "ApplicationMetrics",
    Causes: "Causes",
    Color: "Color",
    HealthStatus: "HealthStatus",
    InstancesHealth: "InstancesHealth",
    RefreshedAt: "RefreshedAt",
    Status: "Status",
};
export const FailureType = {
    CancellationFailed: "CancellationFailed",
    InternalFailure: "InternalFailure",
    InvalidEnvironmentState: "InvalidEnvironmentState",
    PermissionsError: "PermissionsError",
    RollbackFailed: "RollbackFailed",
    RollbackSuccessful: "RollbackSuccessful",
    UpdateCancelled: "UpdateCancelled",
};
export const EventSeverity = {
    DEBUG: "DEBUG",
    ERROR: "ERROR",
    FATAL: "FATAL",
    INFO: "INFO",
    TRACE: "TRACE",
    WARN: "WARN",
};
export const InstancesHealthAttribute = {
    All: "All",
    ApplicationMetrics: "ApplicationMetrics",
    AvailabilityZone: "AvailabilityZone",
    Causes: "Causes",
    Color: "Color",
    Deployment: "Deployment",
    HealthStatus: "HealthStatus",
    InstanceType: "InstanceType",
    LaunchedAt: "LaunchedAt",
    RefreshedAt: "RefreshedAt",
    System: "System",
};
export const EnvironmentInfoType = {
    bundle: "bundle",
    tail: "tail",
};
export const ValidationSeverity = {
    error: "error",
    warning: "warning",
};
