// @amzn/brazil/dist/lib/coverage.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateCoverageDataSync = exports.generateCoverageData = exports.coverageReporters = exports.coverageDataFile = void 0;
const path_1 = __importDefault(require("path"));
const fs_extra_1 = __importDefault(require("fs-extra"));
const workspace_1 = require("./workspace");
const outputDir = path_1.default.join(workspace_1.documentationDir, "coverage");
exports.coverageDataFile = path_1.default.join(workspace_1.generatedMakeDir, "coverage-data.txt");
exports.coverageReporters = [
    {
        filename: "clover.xml",
        regexp: /<metrics statements="(\d+)" coveredstatements="(\d+)" conditionals="(\d+)" coveredconditionals="(\d+)/,
    },
    {
        filename: "cobertura-coverage.xml",
        regexp: /<coverage\s+lines-valid="(\d+)"\s+lines-covered="(\d+).*?branches-valid="(\d+)"\s+branches-covered="(\d+)/,
    },
    {
        filename: "coverage-summary.json",
        regexp: /\{"total": \{"lines":\{"total":(\d+),"covered":(\d+).*?"branches":\{"total":(\d+),"covered":(\d+)/,
    },
];
const defaultArgs = { coverageDir: "coverage", language: "javascript" };
function getArgs(args) {
    return {
        coverageDir: args.coverageDir ?? defaultArgs.coverageDir,
        language: args.language ?? defaultArgs.language,
    };
}
function triples(data, regexp, language) {
    const [, lt, lc, bt, bc] = data.match(regexp).map(Number);
    return (`${language}:line:${(lc / lt) * 100}\n` +
        `${language}:branch:${(bc / bt) * 100}\n`);
}
async function generateCoverageData(args = {}) {
    const { coverageDir, language } = getArgs(args);
    if (await fs_extra_1.default.pathExists(coverageDir)) {
        await fs_extra_1.default.ensureDir(outputDir);
        await fs_extra_1.default.copy(coverageDir, outputDir);
    }
    let data;
    for (const { filename, regexp } of exports.coverageReporters) {
        try {
            data = await fs_extra_1.default.readFile(path_1.default.join(outputDir, filename), "utf8");
        }
        catch (err) {
            continue;
        }
        await fs_extra_1.default.outputFile(exports.coverageDataFile, triples(data, regexp, language), {
            flag: "a",
        });
        break;
    }
}
exports.generateCoverageData = generateCoverageData;
function generateCoverageDataSync(args = {}) {
    const { coverageDir, language } = getArgs(args);
    if (fs_extra_1.default.pathExistsSync(coverageDir)) {
        fs_extra_1.default.ensureDirSync(outputDir);
        fs_extra_1.default.copySync(coverageDir, outputDir);
    }
    let data;
    for (const { filename, regexp } of exports.coverageReporters) {
        try {
            data = fs_extra_1.default.readFileSync(path_1.default.join(outputDir, filename), "utf8");
        }
        catch (err) {
            continue;
        }
        fs_extra_1.default.outputFileSync(exports.coverageDataFile, triples(data, regexp, language), {
            flag: "a",
        });
        break;
    }
}
exports.generateCoverageDataSync = generateCoverageDataSync;
