// @amzn/brazil/dist/lib/workspace.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.buildTools = exports.workspaceRoot = exports.tmpDir = exports.commonJsLibDir = exports.generatedMakeDir = exports.integOutputDir = exports.testOutputDir = exports.documentationDir = exports.packageIdentifier = exports.packageMajorVersion = exports.packageName = exports.privateDir = exports.libDir = exports.buildDir = exports.srcDir = void 0;
const path_1 = __importDefault(require("path"));
const fs_1 = __importDefault(require("fs"));
const cwd = process.cwd();
const { root } = path_1.default.parse(cwd);
function findUp(filename) {
    let dirname = cwd;
    do {
        if (fs_1.default.readdirSync(dirname).indexOf(filename) > -1) {
            return path_1.default.join(fs_1.default.realpathSync(dirname), filename);
        }
    } while (dirname !== root && (dirname = path_1.default.dirname(dirname)));
}
exports.srcDir = path_1.default.dirname(findUp("Config") || "");
exports.buildDir = path_1.default.join(exports.srcDir, "build");
exports.libDir = path_1.default.join(exports.buildDir, "lib");
exports.privateDir = path_1.default.join(exports.buildDir, "private");
exports.packageName = process.env["BRAZIL_PACKAGE_NAME"];
exports.packageMajorVersion = process.env["BRAZIL_PACKAGE_INTERFACE"];
exports.packageIdentifier = `${exports.packageName}-${exports.packageMajorVersion}`;
exports.documentationDir = path_1.default.join(exports.buildDir, "brazil-documentation");
exports.testOutputDir = path_1.default.join(exports.buildDir, "brazil-unit-tests");
exports.integOutputDir = path_1.default.join(exports.buildDir, "brazil-integ-tests");
exports.generatedMakeDir = path_1.default.join(exports.buildDir, "generated-make");
exports.commonJsLibDir = path_1.default.join(exports.libDir, "commonjs");
exports.tmpDir = path_1.default.join(exports.privateDir, "tmp");
exports.workspaceRoot = path_1.default.dirname(findUp("packageInfo") ?? "");
exports.buildTools = process.env["BRAZIL_BUILD_TOOLS"]?.split(path_1.default.delimiter) ?? [];
